concent_api/middleman/queue_operations.py

```diff
@@ -0,0 +1,186 @@
+from asyncio import IncompleteReadError
+from asyncio import Queue
+from asyncio import StreamReader
+from asyncio import StreamWriter
+from collections import OrderedDict  # noqa, pylint: disable=unused-import
+from logging import getLogger
+from logging import Logger
+from typing import Dict
+
+from django.conf import settings
+from django.utils.timezone import now
+
+from middleman.constants import CONNECTION_COUNTER_LIMIT
+from middleman.constants import MessageTrackerItem
+from middleman.constants import RequestQueueItem
+from middleman.constants import ResponseQueueItem
+from middleman_protocol.constants import PayloadType
+from middleman_protocol.exceptions import BrokenEscapingInFrameMiddlemanProtocolError
+from middleman_protocol.exceptions import FrameInvalidMiddlemanProtocolError
+from middleman_protocol.exceptions import PayloadTypeInvalidMiddlemanProtocolError
+from middleman_protocol.exceptions import SignatureInvalidMiddlemanProtocolError
+from middleman_protocol.message import ErrorFrame
+from middleman_protocol.message import GolemMessageFrame  # noqa, pylint: disable=unused-import
+from middleman_protocol.registry import create_middleman_protocol_message
+from middleman_protocol.stream_async import handle_frame_receive_async
+from middleman_protocol.stream_async import map_exception_to_error_code
+from middleman_protocol.stream_async import send_over_stream_async
+
+logger = getLogger()
+
+
+async def request_producer(
+    request_queue: Queue,
+    response_queue: Queue,
+    reader: StreamReader,
+    connection_id: int,
+):
+    while True:
+        try:
+            frame = await handle_frame_receive_async(reader, settings.CONCENT_PUBLIC_KEY)
+        except IncompleteReadError:
+            logger.info(f"Client has closed the connection: {connection_id}. Ending 'request_producer' coroutine")
+            break
+        except (FrameInvalidMiddlemanProtocolError,
+                BrokenEscapingInFrameMiddlemanProtocolError,
+                SignatureInvalidMiddlemanProtocolError,
+                PayloadTypeInvalidMiddlemanProtocolError) as exception:
+            logger.info("Received invalid message")
+            await response_queue.put(
+                ResponseQueueItem(
+                    create_error_frame(exception),
+                    0,  # TODO: update after this constant is added to MiddlemanProtocol
+                    now().timestamp()
+                )
+            )
+            continue
+        logger.info(f"Received message from Concent: request ID = {frame.request_id}, connection ID = {connection_id}")
+        item = RequestQueueItem(
+            connection_id,
+            frame.request_id,
+            frame.payload,
+            now().timestamp()
+        )
+        await request_queue.put(item)
+
+
+async def request_consumer(
+    request_queue: Queue,
+    response_queue_pool: Dict[int, Queue],
+    message_tracker: Dict[int, MessageTrackerItem],
```

Comment:
Shouldn't it be `OrderedDict` too ?

Reply:
Does anyone else gets pylint warnings with this changes?:
```
E: 76,21: Value 'OrderedDict' is unsubscriptable (unsubscriptable-object)
E:103,21: Value 'OrderedDict' is unsubscriptable (unsubscriptable-object)
E:170,21: Value 'OrderedDict' is unsubscriptable (unsubscriptable-object)
```
If yes, and if this is "must have" workaround, let's at lest sweep it under the carpet (xD) with `# pylint: disable`.

Reply:
Please remember our rules on sweeping under the carpet: https://github.com/Code-Poets/coding-style/issues/8